bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/network/SiemensHvacConnectorImpl.java

```diff
@@ -0,0 +1,481 @@
+            @Nullable SiemensHvacBridgeBaseThingHandler hvacBridgeBaseThingHandler) {
+        this.hvacBridgeBaseThingHandler = hvacBridgeBaseThingHandler;
+    }
+
+    public void unsetSiemensHvacBridgeBaseThingHandler(SiemensHvacBridgeBaseThingHandler hvacBridgeBaseThingHandler) {
+        this.hvacBridgeBaseThingHandler = null;
+    }
+
+    public void setBaseUrl(String baseUrl) {
+        this.baseUrl = baseUrl;
+    }
+
+    public void setUserName(String userName) {
+        this.userName = userName;
+    }
+
+    public void setUserPassword(String userPassword) {
+        this.userPassword = userPassword;
+    }
+
+    @Override
+    public void onComplete(@Nullable Request request) {
+        lockObj.lock();
+        try {
+            logger.debug("OnComplete");
+            completedRequest++;
+        } finally {
+            lockObj.unlock();
+        }
+    }
+
+    @Override
+    public void onError(@Nullable Request request, @Nullable SiemensHvacCallback cb) throws Exception {
+        lockObj.lock();
+        try {
+            logger.debug("OnError");
+            completedRequest++;
+        } finally {
+            lockObj.unlock();
+        }
+
+        if (cb == null || request == null) {
+            return;
+        }
+
+        try {
+            final Request retryRequest = httpClient.newRequest(request.getURI());
+            request.method(HttpMethod.GET);
+
+            if (retryRequest != null) {
+                executeRequest(retryRequest, cb);
+            }
+        } catch (Exception ex) {
+            logger.error("exception");
+            throw ex;
+        }
+    }
+
+    private @Nullable ContentResponse executeRequest(final Request request, @Nullable SiemensHvacCallback callback)
+            throws Exception {
+        request.timeout(240, TimeUnit.SECONDS);
+
+        ContentResponse response = null;
+
+        @Nullable
+        SiemensHvacRequestListener requestListener = null;
+        if (callback != null) {
+            requestListener = new SiemensHvacRequestListener(callback, this);
+            request.onResponseSuccess(requestListener);
+            request.onResponseFailure(requestListener);
+        }
+
+        try {
+            if (requestListener != null) {
+                lockObj.lock();
+                try {
+                    startedRequest++;
+                    logger.debug("StartedRequest : {}", startedRequest);
+
+                } finally {
+                    lockObj.unlock();
+                }
+
+                request.send(requestListener);
+            } else {
+                response = request.send();
+            }
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            throw new SiemensHvacException("siemensHvac:Exception by executing request: " + request.getQuery() + " ; "
+                    + e.getLocalizedMessage());
+        }
+        return response;
+    }
+
+    private void _initConfig() throws Exception {
```

Comment:
Weird method name

Reply:
I suppose your concern is about the underscore ?
Remove it.